new_lineage_panel/src/graph.ts

```diff
@@ -472,19 +488,26 @@ export const removeColumnNodes = (
 };
 
 const mergeCollectColumns = (
-  setCollectColumns: Dispatch<SetStateAction<Record<string, string[]>>>,
-  newCollectColumns: Record<string, string[]>
+  setCollectColumns: Dispatch<SetStateAction<Record<string, CollectColumn[]>>>,
+  newCollectColumns: Record<string, CollectColumn[]>
 ) => {
   setCollectColumns((prev) => {
-    const collectColumns: Record<string, string[]> = { ...prev };
+    const collectColumns: Record<string, CollectColumn[]> = { ...prev };
     for (const t in newCollectColumns) {
       const _columns = newCollectColumns[t];
       if (!(t in collectColumns)) {
         collectColumns[t] = _columns;
         continue;
       }
+      // merge _columns with collectColumns[t]
+      const updatedColumns = _columns.map((c) => {
+        const existing = collectColumns[t].findIndex((x) => x.column === c.column);
+        if (existing === -1) return c;
+        if (c.lensType) collectColumns[t][existing].lensType = c.lensType;
+        return null;
```

Comment:
in this case we are basically ignoring column when `lensType` is not there. maybe it would be better to show it with unknown lensType

Reply:
here we are not ignoring the column. We update the lenstype if the column is already existing.